link/uprobe_multi.go

```diff
@@ -0,0 +1,199 @@
+package link
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"unsafe"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/cilium/ebpf/internal"
+	"github.com/cilium/ebpf/internal/sys"
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+// UprobeMultiOptions defines additional parameters that will be used
+// when opening a UprobeMulti Link.
+type UprobeMultiOptions struct {
+	// Path of binary to attach to.
+	Path string
+
+	// Symbol addresses. If set, overrides the address eventually parsed
+	// from the executable.
+	// Mutually exclusive with symbols UprobeMulti argument.
+	Addresses []uint64
+
+	// Offsets into functions provided by symbols array in UprobeMulti
+	// For example to set uprobes to main+5 and _start+10 call UprobeMulti
+	// with:
+	//     symbols: "main", "_start"
+	//     opt.Offset: 5, 10
+	Offsets []uint64
+
+	// Optional, array of associated ref counter offsets.
+	RefCtrOffsets []uint64
+
+	// Optional, array of associated BPF cookies.
+	Cookies []uint64
+}
+
+// addressMulti calculates the array of addresses from symbols array in the executable.
+//
+// opts must not be nil.
+func (ex *Executable) uprobeMultiAddresses(symbols []string, opts *UprobeMultiOptions) ([]uint64, error) {
+	var addresses []uint64
+
+	offsets := len(opts.Offsets)
+	for idx, symbol := range symbols {
+		address, err := ex.address(symbol)
+		if err != nil {
+			return []uint64{0}, err
```

Comment:
Why `[]uint64{0}`? `nil` is more common in such cases.

Reply:
yes, will change